### DISCUSSION TITLE
How to scale the GRF function space

### DISCUSSION THREAD

**Discussion by Nhatkhang** [lululxvi/deepxde#1725] [Q&A]
Hi Prof. Lu, 
I am currently developing a PI-DeepOnet. To create the list of function input for the branch net I use the dde.data.GRF(length_scale=0.1). Then I feed it to the data.PDEOperatorCartesianProd. However, I would like to scale the value of function to be in a specific range for example (-100,100). Could you please guide me how to do that?
Thank you in advance!

**Comment by vl-dud** [comment]
You can define your own class

```
class CustomGRF(dde.data.GRF):
    def _scale(self, values):
        # scale as you wish
        ...

    def eval_one(self, feature, x):
        return self._scale(super().eval_one(feature, x))

    def eval_batch(self, features, xs):
        return self._scale(super().eval_batch(features, xs))
```
